### PR TITLE
[Mailer] Complete mailer documentation

### PR DESCRIPTION
**Pull request by l-vo**
Changes of #14083 reported to 4.4.

### CODE REVIEW

File: reference/configuration/framework.rst [hunk 1]
```diff
@@ -2888,6 +2903,100 @@ Name of the lock you want to create.
...
+        };
+
+.. _mailer-headers:
+
+headers
+.......
+
+**type**: ``array``
+
+Headers to add to emails. key (`name` attribute in xml format) is the header name and value the header value.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
Headers to add to emails. key (``name`` attribute in xml format)
is the header name and value the header value.
```

File: reference/configuration/framework.rst [hunk 2]
```diff
@@ -2888,6 +2903,100 @@ Name of the lock you want to create.
...
+        // config/packages/mailer.php
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+        return static function (ContainerConfigurator $containerConfigurator): void {
+            $containerConfigurator->extension('framework', [
+                'mailer' => [
+                    'dsn' => 'smtp://localhost:25',
+                    'envelope' => [
+                        'recipients' => [
+                            'admin@symfony.com',
+                            'lead@symfony.com'
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
                            'lead@symfony.com',
```

File: reference/configuration/framework.rst [hunk 3]
```diff
@@ -2888,6 +2903,100 @@ Name of the lock you want to create.
...
+
+envelope
+........
+
+sender
+""""""
+
+**type**: ``string``
+
+Sender used by the `Mailer`. Keep in mind that this setting override a sender set in the code.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
Sender used by the ``Mailer``. Keep in mind that this setting override a sender set in the code.
```

File: reference/configuration/framework.rst [hunk 4]
```diff
@@ -2888,6 +2903,100 @@ Name of the lock you want to create.
...
+
+recipients
+""""""""""
+
+**type**: ``array``
+
+Recipients used by the `Mailer`. Keep in mind that this setting override recipients set in the code.
```
Comment (OskarStark) [inline reference/configuration/framework.rst]:
```suggestion
Recipients used by the ``Mailer``. Keep in mind that this setting override recipients set in the code.
```

**Comment by l-vo** [comment]
@OskarStark thank you, suggestions applied and reported to #14083

**Comment by wouterj** [comment]
Thank you @l-vo! I've made some minor formatting fixes in ad317c3, but other than that these changes are perfect!

**Pull request merged**